test/extended/baremetal/hosts.go

```diff
@@ -19,15 +19,48 @@ import (
 	e2eskipper "k8s.io/kubernetes/test/e2e/framework/skipper"
 )
 
-func skipIfNotBaremetal(oc *exutil.CLI) {
+func skipIfNotSupported(oc *exutil.CLI, dc dynamic.Interface) {
 	g.By("checking platform type")
 
 	infra, err := oc.AdminConfigClient().ConfigV1().Infrastructures().Get(context.Background(), "cluster", metav1.GetOptions{})
 	o.Expect(err).NotTo(o.HaveOccurred())
 
-	if infra.Status.PlatformStatus.Type != configv1.BareMetalPlatformType {
-		e2eskipper.Skipf("No baremetal platform detected")
+	provisioningNetwork := getProvisioningNetwork(dc)
```

Comment:
It seems this could be moved down just for the `NonePlatformType` case, no need to run it everytime

Reply:
Yes, I can move it within the switch statement. Please note that the OpenStack and vSpehere cases have a fallthrough which means this will be called for those platforms too.